mmf/trainers/core/training_loop.py

```diff
@@ -95,16 +95,22 @@ def run_training_epoch(self) -> None:
                     combined_report.accumulate_tensor_fields(
                         report, self.metrics.required_params
                     )
+                    # loss needs to be accumulated correctly for metric calculation
+                    combined_report.accumulate_loss(report)
```

Comment:
It would make sense to create a new `accumulate` function which does both as both are needed always. This will make sure both are never decoupled again.